distrepos/mirror_run.py

```diff
@@ -0,0 +1,115 @@
+"""
+This module contains the functions for populating mirrors for a single tag.
+The main entry point is update_mirrors_for_tag(); other functions are helpers.
+"""
+
+import logging
+from distrepos.params import Options, Tag
+from distrepos.tag_run import update_release_repos
+import typing as t
+import socket
+import string
+import os
+import requests
+from datetime import datetime, timedelta
+from pathlib import Path
+
+_log = logging.getLogger(__name__)
+
+def get_baseline_urls() -> t.List[str]:
+    """
+    Get the 
+    """
+    timeout = 5
+    socket.setdefaulttimeout(timeout)
+    if 'repo-itb' in socket.gethostname():
+        return [
+            "http://repo-itb.opensciencegrid.org", 
+            "http://repo-itb.osg-htc.org"
+        ]
+    else:
+        return [
+            "http://repo.opensciencegrid.org", 
+            "http://repo.osg-htc.org"
+        ]
+
+def get_mirror_info_for_arch(hostname: str, tag: Tag, arch: str) -> t.Tuple[str, str]:
+    """
+    Given the top level domain of a potential mirror, find the expected path for that domain
+    that would contain a mirror of the given tag. 
+    """
+    path_arch = string.Template(tag.arch_rpms_mirror_base).safe_substitute({"ARCH": arch})
+    # TODO this might be a misuse of os.path.join. The more appropriate function,
+    # urllib.parse.urljoin, is very sensitive to leading/trailing slashes in the path parts though
+    mirror_base = os.path.join(hostname, path_arch)
+    repomd_url = os.path.join(mirror_base, 'repodata', 'repomd.xml')
```

Comment:
```suggestion
    repomd_url = os.path.join(mirror_base, 'repodata/repomd.xml')
```
Joining constants is just a pet peeve of mine -- makes them harder to grep for.